chain/network/src/network_protocol/edge.rs

```diff
@@ -248,6 +248,13 @@ impl Edge {
             Err(InvalidNonceError::NonceOutOfBoundsError { nonce })
         }
     }
+
+    // Returns a single edge with the highest nonce for each key of the input edges.
+    pub fn deduplicate<'a>(mut edges: Vec<Edge>) -> Vec<Edge> {
+        edges.sort_by_key(|e| e.nonce());
```

Comment:
If we are sorting (n log n) anyway, might just dedup_by: https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.dedup_by

Reply:
done